backend/docs/introducing_new_tools.md

```diff
@@ -0,0 +1,129 @@
+
+As you can see from the `LabelElement` our `tool` column serves as the discriminator, which indicates the type of represented object:
+
+```python
+__mapper_args__ = {
+    'polymorphic_identity': 'LabelElement',
+    'polymorphic_on': tool,
+}
+```
+
+Now let's create our own model!
+
+```python
+class NewToolsLabelElement(LabelElement):
+    """Definition of a Label Element made with New Tool."""
+
+    __tablename__ = 'NewToolLabelElements'
+    id: LabelElementID = Column(String, ForeignKey('LabelElements.id'), primary_key=True)
+
+    # Here provide a set of parameters that are relevant to your NewToolLabelElement.
+
+    __mapper_args__ = {
+        'polymorphic_identity': LabelTool.NewTool,
+    }
+
+    # Your NewToolsLabelElement should also have __init__ and __repr__ methods.
+```
+
+### Step 3. Performing alembic migration
+
+After introducing new model to our database we need to update it. Please refer to [this](changing_database_models.md) tutorial to learn how to do that.
+
+### Step 4. Methods for adding new label
+
+#### Step 4.1 Repository
+
+Create a method in `medtagger/repositories/labels.py` that will create a new `db_session` and add your NewToolLabelElement. Please refer to other examples in the same file.
+
+#### Step 4.2 Business
+
+Create a method in `medtagger/api/scans/business.py` that will call method you have created in the last step. Please refer to other examples in the same file.
+
+Now search for `add_label_element` method in the same file. As you can see this method handless adding new label elements by the tool that the label was made with. Please add your tool and the method that will be responsible for creating such label.
+
+If your tool requires some additional validation. Please write the validation method and add it to `validate_label_payload` in `medtagger/api/scans/business.py`.
+
+#### Step 4.3 REST Service
+
+As you can see in `medtagger/api/scans/service_rest.py` we do already have endpoint that adds new label for given scan. What is now important is to define a schema that will ensure that whatever JSON representation of a label made with your tool is correct.
+
+To define a schema proceed to `medtagger/api/scans/serializers.py`. As you can see we have defined a schema for every `LabelElement` and we are actually telling that our new `LabelElement` can be one of the following types:
+
+```python
+elements_schema = {
+    'type': 'array',
+    "items": {
+        "type": "object",
+        'oneOf': [
+            {'$ref': '#/definitions/rectangular_label_element_schema'},
+            {'$ref': '#/definitions/brush_label_element_schema'},
+            {'$ref': '#/definitions/point_label_element_schema'},
+            {'$ref': '#/definitions/chain_label_element_schema'},
+        ],
+    },
+```
+Below is and example of one of the schemas for label made with rectangle tool:
```

Comment:
1. `Below is and example of one...` -> Did you mean `Below is an example of one...`?
2. Please add a single empty line above this one.